### PR TITLE
AArch64: Implement unsafe helper functions

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements the unsafe helper functions for aarch64.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
The Travis CI build failed with:
```
DeadTreesElimination.cpp:742:39: error: ‘class TR::ILOpCode’ has no member named ‘isPackedExponentiation’
                if (child->getOpCode().isPackedExponentiation())
```
This is not caused by this commit.

**Comment by hangshao0** [comment]
I guess you also need to add vpath to module.xml for AArch64:
https://github.com/eclipse/openj9/blob/master/runtime/vm/module.xml#L148toL150

File: runtime/vm/xr64/unsafeHelper.s [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# Prototype: void unsafePut64(I_64 *address, I_64 value);
+# Defined in: #Args: 2
+	.align	2
+	.global	unsafePut64
+	.type	unsafePut64, %function
+unsafePut64:
+	str	x1, [x0]
+	ret
+	.size	unsafePut64, .-unsafePut64
+# Prototype: void unsafePut32(I_32 *address, I_32 value);
```
Comment (DanHeidinga) [inline runtime/vm/xr64/unsafeHelper.s]: It's a fairly minor thing but can you add a blank line between the different functions being defined in this file?  It makes it little easier to pick out where one ends and the next begins

**Comment by hangshao0** [comment]
There are also changes required in https://github.com/eclipse/openj9/blob/master/runtime/vm/gphandle.c#L238 that handles SIGBUS from the unsafe helpers.

The idea is that when SIGBUS is triggered from accessing the invalid address in the unsafe helpers, we know the UNSAFE_TARGET_REGISTER (x0 here) has the invalid address, then we update this register to an writable/readable but garbage address on stack, so that we can bypass the offending instruction and turn SIGBUS into an exception. 

If you are not making the above change in this PR, please put some comments in unsafeHelper.s and gphandle.c saying that this needs to be done on AArch64.

**Comment by knn-k** [comment]
@hangshao0 Thanks for your comments.

I will add vpath to module.xml.  What should the spec name be for AArch64?  I would say `linux_aarch64`.  I guess I also have to put a new file in openj9/buildspecs with the same name -- Correct?

File: runtime/vm/xr64/unsafeHelper.s (hunk 1 above)
Comment (knn-k) [reply runtime/vm/xr64/unsafeHelper.s]: Sure.

**Comment by hangshao0** [comment]
>What should the spec name be for AArch64? I would say linux_aarch64

I am fine with this name.

> I guess I also have to put a new file in openj9/buildspecs with the same name (I am not going to do that in this PR) -- Correct?

Yes.

**Comment by DanHeidinga** [comment]
Jenkins compile xlinux jdk8

**Comment by DanHeidinga** [comment]
@knn-k can you rebase to resolve the conflict?  I think it should be fairly straightforward

**Comment by knn-k** [comment]
Resolved merge conflict.

**Comment by knn-k** [comment]
Corrected copyright years.

**Comment by DanHeidinga** [comment]
Jenkins compile xlinux jdk8

**Comment by knn-k** [comment]
I would like this to be merged if the code is OK.

**Pull request merged**